src/main/java/com/spotify/github/v3/clients/RepositoryClient.java

```diff
@@ -242,6 +245,56 @@ public CompletableFuture<List<RepositoryInvitation>> listInvitations() {
     return github.request(path, LIST_REPOSITORY_INVITATION);
   }
 
+  /**
+   * Downloads a tar archive of the repository’s default branch (usually main).
+   *
+   * @return a CompletableFuture that resolves to an Optional InputStream
+   */
+  public CompletableFuture<Optional<InputStream>> downloadTarball() {
+    return downloadTarball(null);
+  }
+
+  /**
+   * Downloads a tar archive of the repository. Use :ref to specify a branch or tag to download.
+   *
+   * @return a CompletableFuture that resolves to an Optional InputStream
+   */
+  public CompletableFuture<Optional<InputStream>> downloadTarball(final String ref) {
+    return downloadRepository(REPOSITORY_DOWNLOAD_TARBALL, ref);
+  }
+
+  /**
+   * Downloads a zip archive of the repository’s default branch (usually main).
+   *
+   * @return a CompletableFuture that resolves to an Optional InputStream
+   */
+  public CompletableFuture<Optional<InputStream>> downloadZipball() {
+    return downloadZipball(null);
+  }
+
+  /**
+   * Downloads a zip archive of the repository. Use :ref to specify a branch or tag to download.
+   *
+   * @return a CompletableFuture that resolves to an Optional InputStream
+   */
+  public CompletableFuture<Optional<InputStream>> downloadZipball(final String ref) {
+    return downloadRepository(REPOSITORY_DOWNLOAD_ZIPBALL, ref);
+  }
+
+  private CompletableFuture<Optional<InputStream>> downloadRepository(final String path, final String ref) {
+    final var repoRef = Strings.nullToEmpty(ref);
```

Comment:
Having to handle `null` is not nice and might introduce bugs in future iterations. The obvious goto would be `Optional`. However, it might not always be the nicest in method calls. Furthermore, using it makes the method overloading redundant. 

My suggestion here is to use `Optional` only for your private method: `downloadRepository(final String path, final Optional<String> maybeRef)`. Your `repoRef` could then become something like `maybeRef.orElse("")`.

In methods like `downloadZipball()`, you would then not call the same method with `null` but `downloadRepository(REPOSITORY_DOWNLOAD_ZIPBALL, Optional.empty())`.

Reply:
Yeah sure, I'll change that after lunch :-)